cmd/telemetrygen/internal/logs/logs.go

```diff
@@ -53,8 +52,14 @@ func Start(cfg *Config) error {
 		return fmt.Errorf("'telemetrygen logs' only supports insecure gRPC")
 	}
 
+	endpoint := cfg.Endpoint
+	if host, _, err := net.SplitHostPort(endpoint); err == nil {
+		if u, err := url.Parse(host); err == nil && u.Scheme == "" {
+			endpoint = fmt.Sprintf("passthrough:///%s", endpoint)
+		}
+	}
```

Comment:
Where does `passthrough` come from? I think we should document the format we expect our endpoint argument to have before doing this, why is 'whatever gRPC expects' not good enough?

Reply:
well gRPC expects a scheme, and otherwise complains in logs about the URL to not have one. So that creates confusion from end users. The log says exactly (copied from telemetrygen logs):
```
2023-09-15T08:52:04.545-0700	INFO	grpc@v1.58.0/clientconn.go:489	[core][Channel #3] Channel created	{"system": "grpc", "grpc_log": true}
2023-09-15T08:52:04.545-0700	INFO	grpc@v1.58.0/clientconn.go:1825	[core][Channel #3] original dial target is: "127.0.0.1:64030"	{"system": "grpc", "grpc_log": true}
2023-09-15T08:52:04.545-0700	INFO	grpc@v1.58.0/clientconn.go:1830	[core][Channel #3] dial target "127.0.0.1:64030" parse failed: parse "127.0.0.1:64030": first path segment in URL cannot contain colon	{"system": "grpc", "grpc_log": true}
2023-09-15T08:52:04.545-0700	INFO	grpc@v1.58.0/clientconn.go:1846	[core][Channel #3] fallback to scheme "passthrough"	{"system": "grpc", "grpc_log": true}
2023-09-15T08:52:04.546-0700	INFO	grpc@v1.58.0/clientconn.go:1854	[core][Channel #3] parsed dial target is: {URL:{Scheme:passthrough Opaque: User: Host: Path:/127.0.0.1:64030 RawPath: OmitHost:false ForceQuery:false RawQuery: Fragment: RawFragment:}}	{"system": "grpc", "grpc_log": true}
2023-09-15T08:52:04.546-0700	INFO	grpc@v1.58.0/clientconn.go:1987	[core][Channel #3] Channel authority set to "127.0.0.1:64030"	{"system": "grpc", "grpc_log": true}
```

Note the info log about `parse failed`. This occurs because the endpoint is treated as a URL, and without a scheme, the URL parsing fails. The next line is relevant:
```fallback to scheme "passthrough"	{"system": "grpc", "grpc_log": true}```

So what we're doing in our code is more or less what gRPC ends up doing, but we do it before they get a chance to complain. I will agree it's odd and not great to have to do this, but I guess this is how we fix things.

The `passthrough` scheme is just an internal gRPC reference that goes to a scheme resolver for passthrough that just does nothing to the URL, and therefore moves on. Other resolvers such as `dns` exist. I am not familiar with the gRPC ecosystem.

Reply:
Thanks for all the details. To be clear: I understand the issue, I think it is a confusing log to have and it's bad for end-users.

My main point of confusion is why this is not an issue for opentelemetry-go or grpc-go users. The way I see it, this is an issue that should also be happening to users of ` grpc.DialContext` or of the `otlp[metric/trace]grpc` packages, and I am confused about why is this an issue specific to telemetrygen (at least I have not been able to find users of either library complaining about this in their issue trackers).

I also would prefer to pass the endpoint verbatim to the gRPC library and instead offer a warning to users (something like "please set `passthrough` to avoid grpc-go warnings" or maybe telling users to use [`ipv4`](https://github.com/grpc/grpc/blob/master/doc/naming.md#detailed-design) which also seems like it should work here).

Reply:
I'm not sure. Do we set the zap grpc logger for our own otlp grpc exporter? I think we have configured telemetrygen to be a bit more verbose with grpc, I guess most other applications out there do not enable grpc logging.

Reply:
> Do we set the zap grpc logger for our own otlp grpc exporter? 

We do generally in the Collector (except in tests): https://github.com/open-telemetry/opentelemetry-collector/blob/68dd7d763b599146ce96d3b936f74666af9db757/otelcol/collector.go#L182-L184

> I guess most other applications out there do not enable grpc logging.

Right, I guess that would make sense. I would prefer to add a log here helping the user and report this upstream instead of changing telemetrygen's behavior

Reply:
Ah, look at this:
https://github.com/open-telemetry/opentelemetry-collector/blob/68dd7d763b599146ce96d3b936f74666af9db757/otelcol/internal/grpclog/logger.go#L14C1-L14C1

So how about we set the log level to warn?

Reply:
Oh, right, that also works for me :)